auto-generated/vector-crypto/llvm-api-tests/vaesz.c

```diff
@@ -0,0 +1,27 @@
+// REQUIRES: riscv-registered-target
+// RUN: %clang_cc1 -triple riscv64 -target-feature +v -disable-O0-optnone \
+// RUN:   -emit-llvm %s -o - | opt -S -passes=mem2reg | \
+// RUN:   FileCheck --check-prefix=CHECK-RV64 %s
+
+#include <riscv_vector.h>
+
+vuint32mf2_t test_vaesz_vs_u32mf2(vuint32mf2_t vd, vuint32mf2_t vs2, size_t vl) {
+  return __riscv_vaesz_vs_u32mf2(vd, vs2, vl);
```

Comment:
I guess `vs2` for `.vs` type should always be m1~

Reply:
Not necessarily, if VLEN is small, it may be required to have a large LMUL so VLMAX can at least cover EGS (Element Group Size) and a small LMUL if VLEN is large (if VLEN=256, mf2 is enough for `vs2` to contain at least one element group)

Reply:
Does this mean we need different intrinsics based on the Zvl*b extension that is enabled? Or else we have to define the intrinsics for the worst case which will constrain the register allocator.

Reply:
Got it, thanks!

Reply:
@topperc @kito-cheng , I think we can't determine the minimum luml to use for `vs2` in .td file, if so, what value of `lmul` should we use for vs2?
`vredsum.vs` use `lmul1` for it's scalar operand, should we do the same?

Reply:
As Nicolas said sometimes it needs to be larger than LMUL=1. It needs to be at least the largest LMUL that will cover the EGS at the  smallest VLEN. For SEW=64 the smallest VLEN is 64. For other SEW  the smallest VLEN is 32.

Reply:
So since current instructions that have `.vs` all have `EEW=32` and `EGS=4`, means we should fix the vs2 of those instructions to `lmul 4`?

Reply:
I think the LMUL for the vs2 operand should never be more than the vd operand?

Reply:
The spec says 
` if(LMUL*VLEN < EGW)  then {
      handle_illegal();  // illegal instruction exception
      RETIRE_FAIL
    } else {
`
for all of instructions that have `EGW=128`, `EGS=4` and `SEW=32` constraints.
Does it mean we should only allow those instructions have at least lmul 4?

Reply:
> The spec says 
> 
> ` if(LMUL*VLEN < EGW)  then {
> 
>       handle_illegal();  // illegal instruction exception
> 
>       RETIRE_FAIL
> 
>     } else {
> 
> `
> 
> for all of instructions that have `EGW=128`, `EGS=4` and `SEW=32` constraints.
> 
> Does it mean we should only allow those instructions to have at least lmul 4?

No. We need to provide the intrinsics to support Zvl*b increasing the minimum VLEN.

Maybe we should make the intrinsics unavailable if the min VLEN isn't large enough?

Reply:
Those intrinsics which have `mf2`, `m1`, `m2` should be removed.

Reply:
Yes, it make sense, thanks!!

Reply:
> I think the LMUL for the vs2 operand should never be more than the vd operand? So I guess the LMUL=8 can use LMUL=4, but all the others should use the same LMUL as vd?

Hi @topperc , I think we should enable any of the `lmul` of `vs2` less than or equal to `vd`.
For example, if we use `vint32m2` for `vd`, and we have `zvl64b` enabled, the `vint32m2` is needed for `vs2`. However, if we have just `zvl128b` enabled, then `vint32m1` is enough for `vs2`.

Reply:
Yeah. I agree that's what we should do.